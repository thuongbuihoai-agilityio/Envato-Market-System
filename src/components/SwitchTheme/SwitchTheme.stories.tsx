import { StoryObj, Meta } from '@storybook/react';

import SwitchTheme from '.';

const meta: Meta<typeof SwitchTheme> = {
  title: 'Custom Components/SwitchTheme',
  tags: ['autodocs'],
  component: SwitchTheme,
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;
type Story = StoryObj<typeof SwitchTheme>;

export const Default: Story = {};
