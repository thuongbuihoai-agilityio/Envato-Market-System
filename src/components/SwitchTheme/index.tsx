import { ReactElement, memo } from 'react';
import { ColorMode, useColorMode } from '@chakra-ui/react';

// Components
import { IconButton } from '@app/components';

// Icons
import { DarkTheme, LightTheme } from '@app/components/Icons';

// Themes
import { colors } from '@app/themes/bases/colors';

const SwitchThemeComponent = () => {
  const { colorMode, toggleColorMode } = useColorMode();

  const icons: Record<ColorMode, ReactElement> = {
    light: <LightTheme color={colors.text.primary.default} />,
    dark: <DarkTheme color={colors.common.white} />,
  };

  return <IconButton onClick={toggleColorMode}>{icons[colorMode]}</IconButton>;
};

const SwitchTheme = memo(SwitchThemeComponent);

export default SwitchTheme;
