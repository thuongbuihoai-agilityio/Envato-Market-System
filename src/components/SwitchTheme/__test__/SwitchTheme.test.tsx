import { render, waitFor } from '@testing-library/react';
import SwitchTheme from '..';
import userEvent from '@testing-library/user-event';

// Mocking Chakra UI useColorMode hook
const useColorModeMock = jest.fn();

jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  useColorMode: () => useColorModeMock,
}));

describe('SwitchTheme component', () => {
  const toggleColorModeMock = jest.fn();
  const colorMode = 'light';

  beforeEach(() => {
    useColorModeMock.mockReturnValue({
      colorMode,
      toggleColorMode: toggleColorModeMock,
    });
  });

  it('renders correctly', () => {
    const { container } = render(<SwitchTheme />);

    expect(container).toMatchSnapshot();
  });

  it('calls toggleColorMode when IconButton is clicked', () => {
    const { getByTestId } = render(<SwitchTheme />);
    const iconButton = getByTestId('icon-button-component');

    userEvent.click(iconButton);
    waitFor(() => {
      expect(toggleColorModeMock).toHaveBeenCalled();
    });
  });
});
