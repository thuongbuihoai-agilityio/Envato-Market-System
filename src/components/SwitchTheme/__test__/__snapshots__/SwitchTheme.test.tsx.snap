// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchTheme component renders correctly 1`] = `
<div>
  <div
    class="css-1ve4u7y"
    data-testid="icon-button-component"
  >
    <button
      aria-label="Send email"
      class="chakra-button css-yt1s1n"
      data-testid="icon-button"
      type="button"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="5"
          stroke-width="1.5"
        />
        <path
          d="M12 2V4"
          stroke-linecap="round"
          stroke-width="1.5"
        />
        <path
          d="M12 20V22"
          stroke-linecap="round"
          stroke-width="1.5"
        />
        <path
          d="M20.6602 7L18.9281 8"
          stroke-linecap="round"
          stroke-width="1.5"
        />
        <path
          d="M5.07178 16L3.33973 17"
          stroke-linecap="round"
          stroke-width="1.5"
        />
        <path
          d="M3.33984 7L5.07189 8"
          stroke-linecap="round"
          stroke-width="1.5"
        />
        <path
          d="M18.9282 16L20.6603 17"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </svg>
    </button>
  </div>
</div>
`;
